### ISSUE TITLE
Ligatures that don't merge glyphs together

### ISSUE THREAD

**Issue by nmsmith** [eigilnikolajsen/commit-mono#104]
### TL;DR

It would be nice if arrows were rendered like this:

![Screenshot 2024-03-30 at 10 23 28 am](https://github.com/eigilnikolajsen/commit-mono/assets/909604/956006d6-8b21-41e9-8de3-eedeb4e2520b)

### Full motivation

I'm aware the author of this font is not a fan of "coding ligatures" that merge multiple glyphs into one. I find them quite questionable as well. Oftentimes, when programmers ask for ligatures, I suspect what they _really_ want is just to make a sequence of characters look closer to the symbol that they are supposed to represent.

The basic example is the right arrow `->`, which is used by many programming languages. In many fonts this character combination looks messy: when interpreted as an arrow, it looks badly-drawn.

Most fonts that support ligatures "fix" this issue by offering a ligature that is basically a stretched version of the → character. The main problem with this solution is that it obscures the identity of the underlying characters; it makes them look like a _different_ character. This is especially confusing for people who are new to programming. (People know how to type `-` and `>`, but they don't know how to type an arrow.)

What programmers want is aesthetically-pleasing source code; they don't necessarily want the → character. The [0xProto](https://github.com/0xType/0xProto) font achieves the best of both worlds by offering an arrow ligature that still maintains the identity of `-` and `>`, yet more closely resembles an arrow. Here are all of the font's ligatures:

<img src="https://github.com/eigilnikolajsen/commit-mono/assets/909604/e0397224-24ff-4e90-8bcc-4a1b1a7acc54" width="400"/>

I think it would be cool if CommitMono explored this approach. IMO, it achieves a very good balance between aesthetics and simplicity. By that I mean: it would satisfy programmers who want a font with "nice arrows" (etc.), while also avoiding the dishonesty and confusion associated with fonts that merge glyphs together.

**Comment by eigilnikolajsen** [comment]
Funnily, this was actually how Commit Mono rendered coding ligatures in the early versions, but I changed the rendering to joined symbols due to popular request. Personally, I agree that the integrety of the symbols must always be kept, so the ligatures in Commit Mono are something you switch on intentionally. I doubt I will include an update to mimic the behaviour described above, as I already think symbols like `->` in Commit Mono are pleasing without modifications. Hope you understand!

**Comment by nmsmith** [comment]
Ah ok. That's a shame. IMO more fonts should explore the 0xProto approach. Even Fira Code is [backpedalling](https://github.com/tonsky/FiraCode/issues/1560) on their arrow ligatures.

**Comment by eigilnikolajsen** [comment]
I agree with you, but it seems there's a conflict of interest among developers on this topic.

**Comment by nmsmith** [comment]
Just to be clear, I wasn't proposing that the joined `->` ligature be _replaced_ with a non-joined one. I was merely suggesting that it would be nice to have the non-joined ligatures available via _some_ configuration of the font. 🙂

**Comment by nmsmith** [comment]
(This is an epic font by the way. It’s the best I’ve ever seen. 👏)
